airflow-core/src/airflow/cli/cli_parser.py

```
@@ -60,30 +59,21 @@
 log = logging.getLogger(__name__)
 
 
-for executor_name in ExecutorLoader.get_executor_names(validate_teams=False):
```

I think we need to add a fallback reading of CLIs in old ways - not sure how to do it without impacting performance too much, because we do not want to **always** load executor class. 

This is important for two reasons:

1) someone want to stay with old "community" provider version but use newer airflow
2) someone has their own executor in their 3rd-party provider with CLIs defined

Possibly one way of doing it is simple mapping of the executors -> provider  (for built-in executors we have mapping from the executor to provider package and checking if the provider has "cli" defined in provider info? Also likely we should raise an error if provider has an executor defined with cli commands AND cli command defined in provider info.

---

Good point!

> Possibly one way of doing it is simple mapping of the executors -> provider (for built-in executors we have mapping from the executor to provider package and checking if the provider has "cli" defined in provider info? Also likely we should raise an error if provider has an executor defined with cli commands AND cli command defined in provider info.

Yes, that is the exact same idea for me. It could be done in ProviderManager, and I will add the same validation for AuthManager as well.